qemu/accel/tcg/cputlb.c

```diff
@@ -2147,6 +2147,21 @@ store_helper(CPUArchState *env, target_ulong addr, uint64_t val,
         }
     }
 
+    if (uc->snapshot_level && mr->ram && mr->priority < uc->snapshot_level) {
+        mr = memory_cow(uc, mr, addr & TARGET_PAGE_MASK, TARGET_PAGE_SIZE);
+        if (!mr) {
+            uc->invalid_error = UC_ERR_NOMEM;
```

Comment:
You forget to set `uc->invalid_addr` though I can't remember exactly the location we use it.

Reply:
I'll add this. `uc->invalid_addr` is currently only set but not read.